### ISSUE TITLE
Failing to acquire conversation lock at low loads with socketio channel

### ISSUE THREAD

**Issue by joejuzl** [RasaHQ/rasa#7921]
**Rasa version**: 1.10.x (maybe also newer versions)

**Issue**:

A customer is seeing lots of `Failed to acquire lock` when testing with socketio at low loads. At higher loads (50 users?) this is causing problems and slowing it down.
This is potentially an issue with all channels, that needs to be checked. 
It could also be a different load related problem, that is just being exposed via the failure to acquire the lock.

I think we should try some simple stress tests on:
- the lock store alone
- rasa with socketio channel
- rasa with rest channel

Related issues:
https://forum.rasa.com/t/failed-to-acquire-lock-for-conversation-id-conversation-id-retrying/22412
https://github.com/RasaHQ/rasa/issues/5396

Rasa bench marking:
https://github.com/RasaHQ/rasa-x/issues/1163

**Comment by b-quachtran** [comment]
@joejuzl The issue appears to be socket-io related. When we tried re-producing using rasa-demo, it was able to handle >15 RPS using REST without issue but the lock fails to acquire using Sockets at just 3 RPS.

**Comment by desmarchris** [comment]
@joejuzl do you have any thoughts so far?

**Comment by TyDunn** [comment]
This affects [Bell Canada's production environment](https://rasa-hq.slack.com/archives/C01FF6WU36E/p1614263045029300?thread_ts=1614260016.023600&cid=C01FF6WU36E), so we are going to prioritise refinement

**Comment by wochinge** [comment]
Closed in favor of https://github.com/RasaHQ/rasa/issues/8100 . 

More details can be found [here](https://rasa-hq.slack.com/archives/C36SS4N8M/p1612829674036800).
TLDR:

>  Failed to acquire lock for conversation ID actually just means that we are waiting for the lock to be free. This is completely fine by itself. A possible reason can be that a user is sending another message before the lock was actually freeed. This can happen as socketio is async (in contrast to REST) and hence can send responses to the user before all actions actually finished running (e.g. with REST we only return answers after running all actions). I tried running this with unique user ids (each user just sending one message) and that works fine.
